src/routes/utils.js

```diff
@@ -0,0 +1,24 @@
+import { orderBy } from 'lodash';
+
+const YEAR_SEPARATOR = ', ';
+const DISPLAYED_YEARS_AMOUNT = 3;
+const YEAR_REGEX = /^([1-9][0-9]{0,3})$/;
+
+const isYear = (value) => YEAR_REGEX.test(value);
+
+// eslint-disable-next-line import/prefer-default-export
+export const getFormattedYears = (publications) => {
+  let years = publications?.map(({ dateOfPublication }) => dateOfPublication);
+
+  if (!years?.length) {
+    return '';
+  }
+
+  years = years
+    .filter((year) => isYear(year))
+    .map((year) => parseInt(year, 10));
+  years = orderBy(years, null, ['desc'])
+    .slice(0, DISPLAYED_YEARS_AMOUNT);
+
+  return years.join(YEAR_SEPARATOR);
+};
```

Comment:
Could you please check possible write code something like this?

`export const getFormattedYears = (publications) => {
  const years = publications?.map(({ dateOfPublication }) => dateOfPublication);

  return years?.length
    ? ''
    : years
      .filter((year) => isYear(year))
      .map((year) => parseInt(year, 10))
      .sort(years, null, ['desc']) //
      .slice(0, DISPLAYED_YEARS_AMOUNT)
      .join(YEAR_SEPARATOR);
};`

Reply:
Done.